lib/Cake/Controller/Controller.php

```diff
@@ -785,6 +785,9 @@ public function redirect($url, $status = null, $exit = true) {
 			}
 		}
 
+		if ($status === null) {
+			$status = 302;
+		}
 		if ($status) {
```

Comment:
Those types are all outside the documented types of `int|array|null` so I'm ok with it.